src/modules/battery_resistance_est/battery_resistance_est.cpp

```diff
@@ -0,0 +1,281 @@
+/****************************************************************************
+ *
+ *   Copyright (C) 2020 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#include "battery_resistance_est.hpp"
+
+InternalRes::InternalRes() :
+	ModuleParams(nullptr),
+	WorkItem(MODULE_NAME, px4::wq_configurations::lp_default)
+{
+}
+
+bool InternalRes::init()
+{
+	if (!_battery_sub.registerCallback()) {
+		PX4_ERR("battery status callback registration failed!");
+		return false;
+	}
+
+	_param_est(0) = _param_r_s_init.get();
+	_param_est(1) = (_param_r_t_init.get() + _param_r_s_init.get())/(_param_r_t_init.get()*_param_c_t_init.get());
+	_param_est(2) = 1.0f/(_param_r_t_init.get()*_param_c_t_init.get());
+	_param_est(3) = _param_v_oc_init.get()/(_param_r_t_init.get()*_param_c_t_init.get());
+
+	_voltage_estimation = _param_bat1_v_charged.get()*_param_bat1_n_cells.get(); //assume fully charged?
+
+	_adaptation_gain(0) = 0.0001f;
+	_adaptation_gain(1) = 0.0001f;
+	_adaptation_gain(2) = 0.0001f;
+	_adaptation_gain(3) = 0.0001f;
+
+	inter_res.best_r_internal_est = 0.f;
+
+	return true;
+}
+
+Vector<float,4> InternalRes::extract_ecm_parameters() {
+
+	Vector<float,4> _ecm_params;
```

Comment:
```suggestion
	Vector<float,4> ecm_params;
```
Our usual convention is to prefix the member variables of the class with an underscore; not the other ones.

Reply:
ah I see, thanks!